### PR TITLE
Update jenkins FIPS compatible infra

### PR DESCRIPTION
**Pull request by keithdadkins**
Stop gap role to enable us to keep our existing Jenkins deployments
up-to-date until we migrate to CB. This commit includes needed roles and
updated infra to get our jenkins instances working on fips enabled images.

The update_jenkins.yml playbook will build a new jenkins instance and will
install a groovy init script that will harden and update jenkins upon launch.

A jenkins.sh script to help automate this process has been added as well.
The script is dependent on loading a specific env file. Please get with me
if you wish to use the script.

### Change Details
- Stops using install_jenkins role as it is broken on fips.
- Adds a new update_jenkins role to be used instead.
- Adds a helper script to help build, deploy, and backup jenkins.
- Appends new instructions in the README.md

### Feedback Requested

Glaring security concerns.

### External References
- [BFD-600](https://jira.cms.gov/browse/BFD-600)

- [BFD-xxx](https://jira.cms.gov/browse/BFD-xxx)

### Security Implications
I removed a broken 'salt' repo from yum.repos.d. Note: I did not uninstall Salt, I just removed the broken repo to prevent it from potentially breaking builds.

### CODE REVIEW

File: ops/ansible/playbooks-ccs/roles/update_jenkins/tasks/configure_jenkins.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    while true; do
+      state="$(cat "$stateFile")"
+      case "$state" in
+        UPDATING)
+          sleep 1
+        ;;
+        RESTARTING)
+          sudo systemctl restart jenkins
+          echo "" > "$stateFile"
+        ;;
```
Comment (njdister) [inline ops/ansible/playbooks-ccs/roles/update_jenkins/tasks/configure_jenkins.yml:59]: Correct me if I'm wrong, but couldn't you get caught in an infinite loop if Jenkins doesn't restart in time?

Comment (keithdadkins) [reply ops/ansible/playbooks-ccs/roles/update_jenkins/tasks/configure_jenkins.yml:59]: systemd will timeout (i believe 90 seconds by default)

**Review by jzulim — APPROVED** [review]
FANTASTIC job on this, way to muscle through it, make a significant improvement, allows us to close a POAM and that helper script and readme updates were MUCH needed. Bravo sir.

File: ops/ansible/playbooks-ccs/roles/update_jenkins/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+# The base URL of the custom YUM/APT repo to be used (the portion that's shared
+# by the signing key and packages).
+jenkins_packages_url_base: 'https://pkg.jenkins.io'
+
+# Set to `true` to add a `proxy: _none_` setting for the YUM repo.
+jenkins_packages_repo_yum_disable_proxy: true
+
+jenkins_admin_username: test
+jenkins_admin_password: supersecret
```
Comment (njdister) [inline ops/ansible/playbooks-ccs/roles/update_jenkins/defaults/main.yml:67]: I assume this isn't actually used or just a temporary as it comes online?

File: ops/ansible/playbooks-ccs/roles/update_jenkins/tasks/install_tools.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+- name: Install - jq
+  yum:
+    name: 
+      - jq
+
+- name: Install nfs-utils
+  yum:
+    name:
+      - nfs-utils
```
Comment (njdister) [inline ops/ansible/playbooks-ccs/roles/update_jenkins/tasks/install_tools.yml:63]: I think these yum tasks can be combined.

**Comment by njdister** [comment]
Can you explain the use of tfenv vs tfswitch and why both are needed?  Also stateful/stateless are on 0.12 but mgmt is on 0.13, is that correct?

**Comment by jzulim** [comment]
> Can you explain the use of tfenv vs tfswitch and why both are needed?  Also stateful/stateless are on 0.12 but mgmt is on 0.13, is that correct?

@keithdadkins ?

File: ops/ansible/playbooks-ccs/roles/update_jenkins/defaults/main.yml (hunk 2 above)
Comment (keithdadkins) [reply ops/ansible/playbooks-ccs/roles/update_jenkins/defaults/main.yml:67]: Not used. I copied/pasted this from the previous role.

File: ops/ansible/playbooks-ccs/roles/update_jenkins/tasks/install_tools.yml (hunk 3 above)
Comment (keithdadkins) [reply ops/ansible/playbooks-ccs/roles/update_jenkins/tasks/install_tools.yml:63]: Yup. Also, we should replace `nfs-utils` with `amazon-efs-utils` as we are mounting EFS file systems and I would like to secure it via IAM roles in the future. But again, I copied and pasted this from the previous existing role.

**Comment by keithdadkins** [comment]
> > Can you explain the use of tfenv vs tfswitch and why both are needed?  Also stateful/stateless are on 0.12 but mgmt is on 0.13, is that correct?
> 
> @keithdadkins ?

Not sure why/where tfswitch got in. I use it personally as I prefer it over tfenv, but in the future, I'd like to normalize on one or the other and setup a single point of truth for versions (terraform, ansible, etc).

**Comment by keithdadkins** [comment]
I'm moving this back to draft as we are planning to migrate to hosted jenkins and FIPS has caused too many issues with our build tools. I'm hoping to migrate off of the rhel7 base image and will reopen if that resolves some of our issues.

**Pull request closed without merging**